Formula/premake.rb

```diff
@@ -16,6 +16,8 @@ class Premake < Formula
     sha256 "4b1ce1c63cc3ecca7e195d4c0350fb6f823f659c36ff6c1193fd99023ed25b12" => :yosemite
   end
 
+  deprecate! :date => "June 19, 2020"
```

Comment:
This should be when the software was deprecated by upstream. We can see our depreciation date in git. Sorry for not being clear about that.

Also, it should be formatted to ISO 8601